src/screens/payment-methods/PaymentMethods.tsx

```diff
@@ -150,15 +155,84 @@ export default class PaymentMethods extends BaseAutoRefreshScreen<Props, State>
               count={count}
               limit={limit}
               skip={skip}
-              renderItem={(item: BillingPaymentMethod) => <PaymentMethodComponent paymentMethod={item} navigation={navigation} />}
+              renderItem={(paymentMethod: BillingPaymentMethod) => (
+                <Swipeable
+                  overshootRight={false}
+                  overshootLeft={false}
+                  childrenContainerStyle={style.swiperChildrenContainer}
+                  renderRightActions={() => this.renderSwipeableActions(paymentMethod, style)}>
+                  <PaymentMethodComponent paymentMethod={paymentMethod} navigation={navigation} />
+                </Swipeable>
+              )}
               refreshing={refreshing}
               manualRefresh={this.manualRefresh}
               onEndReached={this.onEndScroll}
-              emptyTitle={i18n.t('paymentMethods.noPaymentMethod')}
+              emptyTitle={I18n.t('paymentMethods.noPaymentMethod')}
             />
           </View>
         )}
       </Container>
     );
   };
+
+  private renderSwipeableActions(paymentMethod: BillingPaymentMethod, style: any) {
+    const deleteInProgress = this.state.deleteOperationsStates?.[paymentMethod.id];
+    const commonColors = Utils.getCurrentCommonColor();
+    return paymentMethod.isDefault ? null : (
+      <TouchableOpacity
+        disabled={deleteInProgress}
+        style={style.trashIconButton}
+        onPress={() => {
+          this.onPressDelete(paymentMethod);
+        }}>
+        {deleteInProgress ? (
+          <ActivityIndicator size={scale(20)} color={commonColors.textColor} />
+        ) : (
+          <Icon style={style.trashIcon} name="trash" />
+        )}
+      </TouchableOpacity>
+    );
+  }
+
+  private onPressDelete(paymentMethod: BillingPaymentMethod): void {
+    Alert.alert(
+      I18n.t('paymentMethods.deletePaymentMethodTitle'),
+      I18n.t('paymentMethods.deletePaymentMethodSubtitle', { cardBrand: paymentMethod.brand, cardLast4: paymentMethod.last4 }),
+      [
+        {
+          text: I18n.t('general.yes'),
+          onPress: () => {
+            this.deletePaymentMethod(paymentMethod.id as string);
+          }
+        },
+        { text: I18n.t('general.cancel') }
+      ]
+    );
+  }
+
+  private async deletePaymentMethod(paymentMethodID: string): Promise<void> {
+    const userID = this.centralServerProvider?.getUserInfo()?.id;
+    this.setState({ deleteOperationsStates: { ...this.state.deleteOperationsStates, [paymentMethodID]: true } });
+    try {
+      // TODO check res data success (waiting for server change)
+      const res = await this.centralServerProvider.deletePaymentMethod(userID, paymentMethodID);
```

Comment:
'res' not needed here ?

Reply:
Now it will be with Claude server-side changes :)